app/app.rb

```diff
@@ -20,16 +20,34 @@ def current_user
     end
 
     def format_date(date)
-    date.strftime("%d/%m/%y")
+      date.strftime("%d/%m/%y")
     end
     
     def reset_date_filter
       session[:date_from] = nil
       session[:date_to] = nil
     end
-  end
 
 
+    def check_booking_from_date(date, range)
+      date.each do |booking|
+        if range.include?(booking.to_s) 
```

Comment:
Check that booking requests for dates entirely outside of (before or after) a space's availability are rejected/not entered into the database.